server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```
@@ -2113,6 +2119,59 @@ public void createSchemaTemplate(CreateTemplatePlan plan) throws MetadataExcepti
     }
   }
 
+  public void appendSchemaTemplate(AppendTemplatePlan plan) throws MetadataException {
+    try {
+      templateManager.appendSchemaTemplate(plan);
+      // write wal
+      if (!isRecovering) {
+        logWriter.appendSchemaTemplate(plan);
+      }
+    } catch (IOException e) {
+      throw new MetadataException(e);
+    }
+  }
+
+  public void pruneSchemaTemplate(PruneTemplatePlan plan) throws MetadataException {
+    try {
+      templateManager.pruneSchemaTemplate(plan);
+      // write wal
+      if (!isRecovering) {
+        logWriter.pruneSchemaTemplate(plan);
+      }
+    } catch (IOException e) {
+      throw new MetadataException(e);
+    }
+  }
+
+  public int countMeasurementsInTemplate(String templateName) throws MetadataException {
+    try {
+      return templateManager.getTemplate(templateName).getMeasurementsCount();
+    } catch (UndefinedTemplateException e) {
+      e.printStackTrace();
+    }
+    return -1;
```

throw exception rather than return -1 or e.printStackTrace()

---

fixed now